text/XXXX-browser-traces.md

```diff
@@ -0,0 +1,98 @@
+
+We became aware of this problem in two ways recently:
+
+1. Users want to sometimes create a transaction within another transaction.  Today there
+   is no way to link these together for the purpose of dynamic sampling.  A separate RFC
+   [0014](https://github.com/getsentry/rfcs/pull/14) is proposed to add an explicit way
+   to carry forward the sampling context for a new transaction started after an already
+   existing one.  It works by explicitly continuing the trace.  This solves part of this
+   issue, but it leaves out the case where the sampling context naturally moves to another
+   Sentry project.
+
+2. We wanted to change our own tracing integration to start tracing on the server
+   [Sentry PR #39349](https://github.com/getsentry/sentry/pull/39349) where this would
+   require mirroring the sampling settings to another Sentry project and would also affect
+   API requests detatched from user sessions.
+
+# Supporting Data
+
+The [honeycomb whitepaper on front-end observability](https://www.honeycomb.io/wp-content/uploads/2022/03/Front-end-Observability-Whitepaper-1.pdf)
+recommends continung traces from the server until the natural end of the user Session:
+
+> To accomplish this task, you will use the first event (page load) as the start of your trace and
+> connect that first event to additional spans to build a full trace of the user session. Each span
+> will represent a single thing that you want to track, such as a server request or a user click.
+
+# Options Considered
+
+There are multiple ways in which this problem can be addressed.
+
+## Encouraged Root Trace Project and Session Long Traces
+
+In the most trivial case the recommendation to customers would be to pick one project that
+starts traces for real user sessions.  This could be *either* frontend of backend but it
+should attempt to be consistent about it.  In either case the client SDK should *continue the
+trace* until the browser tab naturally closes.
+
+The consequences are that ``startTransaction()`` always anchors to the already open trace
+and dynamic sampling context.  There can be an extra flag to force the start of a new trace
+but that would be strongly recommended against.
+
+## Alternative A: Detaching Sampling Project from Root Project
+
+An alternative approach would be to allow a transaction to start again on the client but to
+continue with the sampling context that came from the server.  In that case the root of the
+trace is in fact the frontend for continued transactions after a page navigate, however the
+sampling context is reused from the original server side request.
+
+In this case the relationship of root project setting the dynamic sampling context would be
+broken up and instead a transaction can explicitly pick up the sampling context of another
+project but still issue disconnected traces.
+
+## Alternative B: Trace to Trace Relationships
+
+A potential alternative would be to continue the current project but allow a trace to annotate
```

Comment:
As an aside, in my eyes there are various correlation ID mechanisms that we chat about, and it would be nice to get an RFC going that establishes them all. Off the top of my head:

- release id
- user id
- session id
- trace id
- profile id
- replay id (basically a session id)

The most important relationships are as follows:
- 1 users x R releases
- 1 user x S sessions/replays
- 1 user x T traces
- 1 session/replay x T traces
- 1 profile x 1 trace (though this is changing to 1 profile x T trace iirc)

If we can establish these concepts consistently it'll be a nice foundation for us to keep building off of

Reply:
big +1 here, i still feel like we are moving torwards treating a trace more like a session, and it will get more and more confusing with our existing sessions/releases and session replay (which is closes to what is the standard definition of a session)

The problem we are solving for to relate transactions across a user session by exanding the inclusion of more traces in a single trace for sampling, I feel will just create more fuzzy area. Confusion for us and our users.